src/Microsoft.ML.ImageAnalytics/ImageLoader.cs

```diff
@@ -91,7 +91,11 @@ internal sealed class Options : TransformInputBase
         internal ImageLoadingTransformer(IHostEnvironment env, string imageFolder = null, params (string outputColumnName, string inputColumnName)[] columns)
             : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(ImageLoadingTransformer)), columns)
         {
-            ImageFolder = imageFolder;
+            // Throws ArgumentException if given imageFolder path is invalid or empty. Note: imageFolder may be null in this case.
+            if (imageFolder != null)
+                ImageFolder = Path.GetFullPath(imageFolder);
+            else
```

Comment:
The two functions are almost identical. Can you have the code inside the function in only one place and have one constructor call the other?

Reply:
Good suggestion, done!